client/src/components/Ticketing/ticketingmanager/SubscriptionRedemption/SubscriptionsRedemptionContainer.tsx

```diff
@@ -0,0 +1,225 @@
+import {
+  TextField,
+  Autocomplete,
+  Select,
+  MenuItem,
+  InputLabel,
+  FormControl,
+} from '@mui/material';
+import React, {useEffect, useState} from 'react';
+import {
+  getSeasonFilterFunction,
+  getSubscriptionFilterFunction,
+  getSubscriptionSortFunction,
+  Subscription,
+  useFetchSeasons,
+} from './SubscriptionRedemptionUtils';
+import {SubscriptionRedemptionContainer} from './SubscriptionRedemptionContainer';
+import {getData} from '../Event/components/ShowingUtils';
+
+interface SubscriptionsRedemptionProps {
+  token: string;
+  setPopUpProps: any;
+  showPopUp: boolean;
+}
+
+export const SubscriptionsRedemptionContainer = (
+  props: SubscriptionsRedemptionProps,
+) => {
+  const {token, showPopUp, setPopUpProps} = props;
+  const {seasons} = useFetchSeasons();
+  const [subscriptions, setSubscriptions] = useState<Subscription[]>([]);
+  const [inputValue, setInputValue] = useState<string>('');
+  const [queries, setQueries] = useState({
+    seasons: [],
+    name: '',
+    email: '',
+  });
+  const [filters, setFilters] = useState({
+    subscriptionFilter: 'all',
+    seasonFilter: 'present',
+    sort: 'type',
+  });
+  const [editing, setEditing] = useState(false);
+  const [search, setSearch] = useState(false);
+
+  useEffect(() => {
+    const controller = new AbortController();
+    if (token) {
+      const seasonQueries = queries.seasons.map(
+        (season) => `seasonid=${season.id}`,
+      );
+
+      const nameQueries = queries.name
+        .trim()
+        .split(' ')
+        .map((name) => `name=${name}`);
+
+      getData(
+        `${
+          process.env.REACT_APP_API_2_URL
+        }/subscription-types/subscriptions/search?${seasonQueries.join(
+          '&',
+        )}&${nameQueries.join('&')}&${
+          queries.email ? `email=${queries.email.trim()}` : ''
+        }`,
```

Comment:
It looks like when you just input an email into the search area without any names, a url like `/search?&name=&email=test@test.com` is created which ends up returning all results. I think if no name is input, then the `&name=` should be omitted.

Reply:
Fixed